File: kotlin-electron/karakum/plugins/convertElectronQualifiedName.js
import ts from "typescript";

export default function (node, context, render) {
    if (
        ts.isQualifiedName(node)
        && ts.isIdentifier(node.left)
        && node.left.text === "Electron"
    ) {
        return `electron.core.${render(node.right)}`
    }

    if (
        ts.isPropertyAccessExpression(node)
        && ts.isIdentifier(node.expression)
        && node.expression.text === "Electron"
    ) {
        return `electron.core.${render(node.name)}`
    }

    return null
}

File: kotlin-electron/karakum/src/jsMain/kotlin/electron/karakum/plugins/convertElectronQualifiedName.kt
package electron.karakum.plugins

import electron.karakum.util.nullable
import io.github.sgrishchenko.karakum.extension.createPlugin
import typescript.isIdentifier
import typescript.isPropertyAccessExpression
import typescript.isQualifiedName

val convertElectronQualifiedName = createPlugin { node, _, render ->
    nullable {
        ensure(isQualifiedName(node))

        val left = node.left
        ensure(isIdentifier(left))
        ensure(left.text == "Electron")

        "electron.core.${render(node.right)}"
    } ?: nullable {
        ensure(isPropertyAccessExpression(node))

        val expression = node.expression
        ensure(isIdentifier(expression))
        ensure(expression.text == "Electron")

        "electron.core.${render(node.name)}"
    }
}
